server/controllers/session-auth.js

```diff
@@ -0,0 +1,96 @@
+const {
+  fetchSpotifyToken,
+  fetchTwitchToken,
+  fetchTwitchChannel,
+} = require("../auth");
+
+const {
+  spotifyClientId,
+  spotifyClientSecret,
+  twitchClientSecret,
+  twitchClientId,
+} = require("../config/credentials.json");
+
+const SessionAuthController = {
+  postClientTwitchAccessCode(req, res) {
+    const {
+      body: { accessKey },
+    } = req;
+
+    fetchTwitchToken(twitchClientId, twitchClientSecret, accessKey)
+      .then((twitchResponse) => {
+        const {
+          access_token,
+          expires_in,
+          refresh_token,
+          token_type,
+        } = twitchResponse;
+
+        req.session.accessKeys.twitchToken = {
+          access_token,
+          expires_in,
+          refresh_token,
+          token_type,
+          expirationDate: new Date(new Date().getTime() + expires_in * 1000),
```

Comment:
I don't think we need to know the date of expiration. We can drop this and the `expires_in` before persisting.

Reply:
word, I'll kill it